src/util-ebpf.c

```diff
@@ -313,141 +497,196 @@ int EBPFSetupXDP(const char *iface, int fd, uint8_t flags)
+                    f->sp, f->dp);
+        FlowCounters *fc = FlowGetStorageById(f, GetFlowBypassCounterID());
+        if (fc == NULL) {
+            FLOWLOCK_UNLOCK(f);
+            flowstats->count++;
+            return true;
+        }
+        if (flow_key->sp == f->sp) {
+            if (pkts_cnt != fc->todstpktcnt) {
+                flowstats->packets += pkts_cnt - fc->todstpktcnt;
+                flowstats->bytes += bytes_cnt - fc->todstbytecnt;
+                fc->todstpktcnt = pkts_cnt;
+                fc->todstbytecnt = bytes_cnt;
+                /* interval based so no meaning to update the millisecond.
+                 * Let's keep it fast and simple */
+                f->lastts.tv_sec = ctime->tv_sec;
+            }
+        } else {
+            if (pkts_cnt != fc->tosrcpktcnt) {
+                flowstats->packets += pkts_cnt - fc->tosrcpktcnt;
+                flowstats->bytes += bytes_cnt - fc->tosrcbytecnt;
+                fc->tosrcpktcnt = pkts_cnt;
+                fc->tosrcbytecnt = bytes_cnt;
+                f->lastts.tv_sec = ctime->tv_sec;
+            }
+        }
+        FLOWLOCK_UNLOCK(f);
+        return false;
+    } else {
+        /* Flow has disappeared so it has been removed due to timeout.
+         * This means we did not see any packet since bypassed_timeout/flow_dump_interval
+         * checks so we are highly probably have no packet to account.
+         * So we just discard the flow */
+        SCLogDebug("No flow for %d -> %d", flow_key->sp, flow_key->dp);
+        SCLogDebug("Dead with pkts %lu bytes %lu", pkts_cnt, bytes_cnt);
+        flowstats->count++;
+        return true;
     }
-    return 0;
 }
 
+typedef bool (*OpFlowForKey)(struct flows_stats *flowstats, FlowKey *flow_key,
+                            uint32_t hash, struct timespec *ctime,
+                            uint64_t pkts_cnt, uint64_t bytes_cnt);
+
 /**
  * Bypassed flows cleaning for IPv4
  *
  * This function iterates on all the flows of the IPv4 table
  * looking for timeouted flow to delete from the flow table.
  */
-static int EBPFForEachFlowV4Table(LiveDevice *dev, const char *name,
+static int EBPFForEachFlowV4Table(ThreadVars *th_v, LiveDevice *dev, const char *name,
                                   struct flows_stats *flowstats,
-                                  struct timespec *ctime)
+                                  struct timespec *ctime,
+                                  struct ebpf_timeout_config *tcfg,
+                                  OpFlowForKey EBPFOpFlowForKey
+                                  )
 {
     int mapfd = EBPFGetMapFDByName(dev->dev, name);
     struct flowv4_keys key = {}, next_key;
     int found = 0;
     unsigned int i;
-    unsigned int nr_cpus = UtilCpuGetNumProcessorsConfigured();
-    if (nr_cpus == 0) {
-        SCLogWarning(SC_ERR_INVALID_VALUE, "Unable to get CPU count");
+    uint64_t hash_cnt = 0;
+
+    if (tcfg->cpus_count == 0) {
```

Comment:
seems like something that should be checked at init

Reply:
Removing the warning and keeping the sanity check as the function is not called often.